overlord/snapstate/snapmgr.go

```diff
@@ -531,6 +531,18 @@ func (snapst *SnapState) InstanceName() string {
 	return snap.InstanceName(cur.RealName, snapst.InstanceKey)
 }
 
+// RefreshInhibitProceedTime is the time after which a pending refresh is forced
+// for a running snap in the next auto-refresh.
+//
+// If nil is returned, then there are no pending refreshes.
+func (snapst *SnapState) RefreshInhibitProceedTime() *time.Time {
+	if snapst.RefreshInhibitedTime == nil {
+		return nil
+	}
+	proceedTime := snapst.RefreshInhibitedTime.Add(maxInhibition)
```

Comment:
not a blocker but we need to keep in mind that this is coming https://github.com/snapcore/snapd/pull/13485,  so we might need state as well here even if we don't use it yet and a TODO

Reply:
good point, updated!

Reply:
this is my suggestion:

```
func RefreshInhibitProceedTime(st *State) func(*snapstate.SnapState) time.Time {
    // assume we have the lock
    ... get the cureMaxInhibition from the  config options ...
    return func(snapst *SnapState) *time.Time {
         // this itself doesn't need to lock
         if snapst.RefreshInhibitedTime == nil {
		return nil
	}
	proceedTime := snapst.RefreshInhibitedTime.Add(curMaxInhibition)
        return &proceedTime
    }
}
```